ml-proto/test/block.wast

```diff
@@ -1,35 +1,165 @@
+
 (module
-  (func $empty
+  (func "empty"
     (block)
     (block $l)
   )
 
-  (func $singular (result i32)
+  (func "singular" (result i32)
+    (block (nop))
     (block (i32.const 7))
   )
 
-  (func $multi (result i32)
-    (block (i32.const 5) (i32.const 6) (i32.const 7) (i32.const 8))
+  (func $nop)
+  (func "multi" (result i32)
+    (block (call $nop) (call $nop) (call $nop) (call $nop))
+    (block (call $nop) (call $nop) (call $nop) (i32.const 8))
+  )
+
+  (func "nested" (result i32)
+    (block (block (call $nop) (block) (nop)) (block (call $nop) (i32.const 9)))
+  )
+
+  (func "deep" (result i32)
+    (block (block (block (block (block (block (block (block (block (block
+      (block (block (block (block (block (block (block (block (block (block
+        (block (block (block (block (block (block (block (block (block (block
+          (block (block (block (block (block (block (block (block (block (block
+            (block (block (block (block (block (call $nop) (i32.const 150))))))
+          ))))))))))
+        ))))))))))
+      ))))))))))
+    ))))))))))
+  )
+
+  (func "unary-operand" (result i32)
+    (i32.ctz (block (call $nop) (i32.const 13)))
+  )
+  (func "binary-operand" (result i32)
+    (i32.mul (block (call $nop) (i32.const 3)) (block (call $nop) (i32.const 4)))
+  )
+  (func "test-operand" (result i32)
+    (i32.eqz (block (call $nop) (i32.const 13)))
+  )
+  (func "compare-operand" (result i32)
+    (f32.gt (block (call $nop) (f32.const 3)) (block (call $nop) (f32.const 3)))
+  )
+
+  (func "br-bare" (result i32)
+    (block (br 0) (unreachable))
+    (i32.const 19)
+  )
+  (func "br-value" (result i32)
+    (block (br 0 (i32.const 18)) (i32.const 19))
+  )
+  (func "br-repeated" (result i32)
+    (block
+      (br 0 (i32.const 18))
+      (br 0 (i32.const 19))
+      (br 0 (i32.const 20))
+      (i32.const 21)
+    )
+  )
+  (func "br-inner" (result i32)
+    (block
+      (block (br 1 (i32.const 22)))
+      (block (br 0))
+      (i32.const 21)
+    )
+  )
+
+  (func "drop-inner" (result i32)
+    (block (call $fx) (i32.const 7) (call $nop) (i32.const 8))
+  )
+  (func "drop-last"
+    (block (call $nop) (call $fx) (nop) (i32.const 8))
+  )
+  (func "drop-br-void"
+    (block (br 0 (nop)))
+    (block (br 0 (call $nop)))
+  )
+  (func "drop-br-value"
+    (block (br 0 (i32.const 8)))
+  )
+  (func "drop-br-value-heterogeneous"
+    (block (br 0 (i32.const 8)) (br 0 (f64.const 8)) (br 0 (f32.const 8)))
+    (block (br 0 (i32.const 8)) (br 0) (br 0 (f64.const 8)))
+    (block (br 0 (i32.const 8)) (br 0 (call $nop)) (br 0 (f64.const 8)))
+    (block (br 0 (i32.const 8)) (br 0) (br 0 (f32.const 8)) (i64.const 3))
+    (block (br 0) (br 0 (i32.const 8)) (br 0 (f64.const 8)) (br 0 (nop)))
+    (block (br 0) (br 0 (i32.const 8)) (br 0 (f32.const 8)) (i64.const 3))
+    (block (block (br 0) (br 1 (i32.const 8))) (br 0 (f32.const 8)) (i64.const 3))
   )
 
-  (func $effects (result i32)
+  (func "effects" $fx (result i32)
```

Comment:
why should this be valid? i don't understand what two names for a function means